File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.java
class Solution {

     public int specialArray(int[] nums) {
        int begin = 0;
        int last = nums.length;
        int ans = -1;
        
        while(begin <= last) {
            int mid = begin + (last-begin)/2;
            
            int count = 0;
            for (int i = 0; i < nums.length ; i++) {
                if (nums[i] >= mid) {
                    count++;
                }
            }
            if (mid == count) {
                ans = mid;
                begin = mid + 1;
            } else if (mid > count) {
                last = mid - 1;
            } else {
                begin = mid + 1;
            }
        }
        
        return ans;
    }
}

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        
        left = 0
        right = len(nums)
        
        while left <= right:
            mid = (left+right)//2
            
            count = 0
            for num in nums:
                if num >= mid: count+=1
            if mid == count:
                return mid
            elif mid < count:
                left = mid+1
            else:
                right = mid-1
                
        return -1
        